src/main/java/com/blackduck/integration/detect/lifecycle/run/operation/OperationRunner.java

```diff
@@ -1596,16 +1603,35 @@ public DetectConfigurationFactory getDetectConfigurationFactory() {
         return this.detectConfigurationFactory;
     }
     
-    public String computeMD5Base64(File file) throws IOException {
-        try (FileInputStream fis = new FileInputStream(file)) {
-            MessageDigest md = DigestUtils.getMd5Digest();
-            byte[] buffer = new byte[8192];
-            int bytesRead;
-            while ((bytesRead = fis.read(buffer)) != -1) {
-                md.update(buffer, 0, bytesRead);
+    private void zipFileIfNecessaryAndComputeMD5Base64(File file, ScassScanInitiationResult initResult, File outputDirectory) throws IOException {
+        MessageDigest md = DigestUtils.getMd5Digest();
+        Encoder encoder = Base64.getEncoder();
+
+        try (ZipFile zipFile = new ZipFile(file)) {
+            try (FileInputStream fis = new FileInputStream(file)) {
```

Comment:
I believe you could do

try (ZipFile zipFile = new ZipFile(file);
  FileInputStream fis = new FileInputStream(file))

to get the auto resource closure but only have 1 try block